libexec/tfenv-resolve-version

```diff
@@ -127,6 +127,30 @@ if [[ "${version_requested}" =~ ^min-required$ ]]; then
   version_requested="${min_required}";
 fi;
 
+if [[ "${version_requested}" =~ ^latest-allowed$ ]]; then
+  log 'debug' 'Detecting latest allowable version...';
+  version_spec="$(grep -h required_version "${TFENV_DIR:-$(pwd)}"/{*.tf,*.tf.json} 2>/dev/null | rev | cut -d'"' -f2 | rev | cut -d, -f1)";
+  version_num="$(echo "${version_spec}" | sed -E 's/[^0-9.]+//')";
+  log 'debug' "Using ${version_num} from version spec: ${version_spec}";
+
+  case "${version_spec}" in
+    '>'*)
+      version_requested=latest;
+      ;;
+    '<='*)
+      version_requested="${version_num}";
+      ;;
+    '~>'*)
+      version_without_rightmost="$(echo "${version_num}" | sed 's/\./\n./g' | head -n -1 | xargs -I@ echo -n @)";
```

Comment:
Seems we can't use a negative count to (BSD) `head` on macOS.

I saw `tfenv` already depends on GNU `grep` on macOS (ed6f2a13cc97ed2b5db11000f1a9b3021be4685e) - do you want to add the `head` (`brew install coreutils`) dependency as well, or shall I find another way to do this?

Reply:
You can get the first line with `sed` which is already a dependency: `sed -n 1p`

Reply:
Here I don't want (only) the first line, but the first `N-1` lines.

e.g. from `1.2.3` we want `1.2`; from `1.2`, `1`.

Of course there's other ways of doing it (perhaps not as concisely?) I just wondered whether to bother.

Reply:
Whoops! Totally misread what you were doing :) It's possible with sed or awk and effort has gone into not adding coreutils as a dependency, so I think it's worth doing.